config/express.js

```diff
@@ -17,63 +24,58 @@ module.exports = (app, config, passport) => {
     app.use(Raven.errorHandler());
   }
   app.use(
-    express.compress({
+    compression({
       filter: function(req, res) {
         return /json|text|javascript|css/.test(res.getHeader("Content-Type"));
       },
       level: 9
     })
   );
-  app.use(express.favicon());
+  // app.use(favicon());
   app.use(express.static(config.root + "/public"));
 
   if (process.env.NODE_ENV !== "test") {
-    app.use(express.logger("dev"));
+    // Use morgan logger if not in a development environment
+    app.use(morgan("dev"));
   }
-  app.configure("development", () => {
-    app.use(express.errorHandler());
+
+  if (process.env.NODE_ENV === "development") {
+    app.use(errorHandler());
     app.locals.pretty = true;
-  });
+  }
 
   app.set("views", config.root + "/app/views");
   app.set("view engine", "pug");
-  app.configure(() => {
-    app.use(helpers(config.app.name));
-    app.use(express.cookieParser());
-    app.use(bodyParser.urlencoded({
-      extended: true
-    }));
-    app.use(bodyParser.json());
-    app.use(express.methodOverride());
-    app.use(
-      express.session({
-        secret: "noobjs",
-        store: new mongoStore({
-          url: config.db,
-          collection: "sessions"
-        })
-      })
-    );
 
-    app.use(flash());
-    app.use(passport.initialize());
-    app.use(passport.session());
-    app.use(app.router);
+  app.use(helpers(config.app.name));
+  app.use(cookieParser());
+  app.use(bodyParser.urlencoded({
+    extended: true
+  }));
+  app.use(bodyParser.json());
+  app.use(methodOverride());
+  app.use(
+    session({
+      secret: "noobjs",
+      resave: false,
+      saveUninitialized: false,
+      store: new mongoStore({
+        url: config.db,
+        collection: "sessions"
+      })
+    })
+  );
 
-    app.use((err, req, res, next) => {
-      if (err.message.indexOf("not found") !== -1) {
-        return next();
-      }
-      console.log(err.stack);
+  app.use(flash());
+  app.use(passport.initialize());
+  app.use(passport.session());
 
-      res.status(500).render("pages/500", { error: err.stack });
-    });
+  app.use((err, req, res, next) => {
+    if (err.message.indexOf("not found") !== -1) {
+      return next();
+    }
+    console.log(err.stack);
 
-    app.use((req, res) => {
```

Comment:
Any reason to remove the render of 404 page on not found routes?

Reply:
Ya, the 404 page was always being rendered when it was included in express.js, so I had to move it to the bottom of routes.js. I think it has to do with not using app.router (deprecated) anymore.

Reply:
Okay!